lib/Common/Memory/HeapBlock.h

```diff
@@ -221,31 +257,58 @@ class HeapBlock
     enum HeapBlockType : byte
     {
         FreeBlockType = 0,                  // Only used in HeapBlockMap.  Actual HeapBlock structures should never have this.
-        SmallNormalBlockType = 1,
-        SmallLeafBlockType = 2,
-        SmallFinalizableBlockType = 3,
+        SmallNormalBlockType,
+        SmallLeafBlockType,
+        SmallFinalizableBlockType,
 #ifdef RECYCLER_WRITE_BARRIER
-        SmallNormalBlockWithBarrierType = 4,
-        SmallFinalizableBlockWithBarrierType = 5,
+        SmallNormalBlockWithBarrierType,
+        SmallFinalizableBlockWithBarrierType,
 #endif
-        MediumNormalBlockType = 6,
-        MediumLeafBlockType = 7,
-        MediumFinalizableBlockType = 8,
+#ifdef RECYCLER_VISITED_HOST
+        SmallRecyclerVisitedHostBlockType,
+#endif
+        MediumNormalBlockType,
+        MediumLeafBlockType,
+        MediumFinalizableBlockType,
 #ifdef RECYCLER_WRITE_BARRIER
-        MediumNormalBlockWithBarrierType = 9,
-        MediumFinalizableBlockWithBarrierType = 10,
+        MediumNormalBlockWithBarrierType,
+        MediumFinalizableBlockWithBarrierType,
+#endif
+#ifdef RECYCLER_VISITED_HOST
+        MediumRecyclerVisitedHostBlockType,
 #endif
-        LargeBlockType = 11,
+        LargeBlockType,
 
-        SmallAllocBlockTypeCount = 6,  // Actual number of types for blocks containing small allocations
-        MediumAllocBlockTypeCount = 5, // Actual number of types for blocks containing medium allocations
-        SmallBlockTypeCount = 11,      // Distinct block types independent of allocation size using SmallHeapBlockT
+#ifdef RECYCLER_VISITED_HOST
+        SmallAllocBlockTypeCount = 7, // Actual number of types for blocks containing small allocations
```

Comment:
There's no test for this today IIRC but you probably need to update JD to handle this update?

Reply:
Yes, I believe it has been updated, but that code lives outside of ChakraCore and isn't part of this PR, or is there a spot under core that needs to be updated?